librz/util/lib.c

```diff
@@ -347,38 +328,84 @@ RZ_API int rz_lib_open_ptr(RzLib *lib, const char *file, void *handler, RzLibStr
 				int minor = dot ? atoi(dot + 1) : 0;
 				// The pkgname member was introduced in 4.2.0
 				if (major > 4 || (major == 4 && minor >= 2)) {
-					printf("rz-pm -ci %s\n", stru->pkgname);
+					eprintf("rz-pm -ci %s\n", stru->pkgname);
 				}
 			}
 			return -1;
 		}
 	}
-	RzLibPlugin *p = RZ_NEW0(RzLibPlugin);
-	p->type = stru->type;
-	p->data = stru->data;
-	p->file = strdup(file);
+	p = RZ_NEW0(RzLibPlugin);
+	if (p == NULL) {
+		eprintf("Library plugin allocation error for file %s\n", file);
+		return -1;
+	}
+	memset (p, 0, sizeof(*p));
+	if ((p->file = strdup(file)) == NULL) {
+		goto err;
+	}
 	p->dl_handler = handler;
 	p->handler = rz_lib_get_handler(lib, p->type);
+	p->type = stru->type;
+	p->data = stru->data;
 	p->free = stru->free;
-
-	int ret = rz_lib_run_handler(lib, p, stru);
-	if (ret == -1) {
+	if (rz_lib_run_handler(lib, p, stru)) {
 		IFDBG eprintf("Library handler has failed for '%s'\n", file);
-		free(p->file);
-		free(p);
-		rz_lib_dl_close(handler);
-	} else {
-		rz_list_append(lib->plugins, p);
+		goto err_file;
 	}
+	rz_list_append(lib->plugins, p);
+	return 0;
+err_file:
+	free (p->file);
+err:
+	RZ_FREE(p);
```

Comment:
you can free normally since `RZ_FREE` frees and set the pointer to NULL, but here you are going to return another value